saleor/graphql/attribute/types.py

```diff
@@ -655,34 +655,33 @@ class Meta:
         doc_category = DOC_CATEGORY_ATTRIBUTES
 
 
-class SelectedAttribute(graphene.Interface):
+class SelectedAttribute(ChannelContextTypeForObjectType):
     attribute = graphene.Field(
         Attribute,
         default_value=None,
         description=AttributeDescriptions.NAME,
         required=True,
     )
     values = NonNullList(
-        AttributeValue,
-        description="Values of an attribute.",
-        required=True,
-        deprecation_reason="Use type-specific fields instead.",
+        AttributeValue, description="Values of an attribute.", required=True
     )
 
     class Meta:
         doc_category = DOC_CATEGORY_ATTRIBUTES
-        description = "Represents an assigned attribute to an object."
+        description = "Represents a custom attribute."
 
-    @staticmethod
-    def resolve_type(instance: SelectedAttributeData, _info):
-        if instance.attribute.node.input_type == AttributeInputType.SINGLE_REFERENCE:
-            entity_type = cast(str, instance.attribute.node.entity_type)
-            return ASSIGNED_SINGLE_REFERENCE_MAP.get(entity_type)
-        if instance.attribute.node.input_type == AttributeInputType.REFERENCE:
-            entity_type = cast(str, instance.attribute.node.entity_type)
-            return ASSIGNED_MULTI_REFERENCE_MAP.get(entity_type)
-        attr_type = ASSIGNED_ATTRIBUTE_MAP[instance.attribute.node.input_type]
-        return attr_type
+
+class AssignedAttributeInterface(graphene.Interface):
```

Comment:
Is the `Interface` word necessary as part of the name? It would look nicer in the schema if you had just `AssignedAttribute`, or is this name already taken?

Reply:
@maarcingebala  `AssignedAttribute` is the name for the `Union` which wraps all possible `AssignedXAttributes`. So I can't use it. 
Do you have any better name for that?